### DISCUSSION TITLE
Error (failed with exit code 1) in GitHub Action check

### DISCUSSION THREAD

**Discussion by philippnurullin** [fonttools/fontbakery#3136] [Q&A] [answered]
## Observed behaviour

After pushing the update, the files are not passing the Action check (Fontbakery CI — variable fonts and Fontbakery CI — static fonts)

`Error: font-bakery Action failed during fontbakery execution attempt with error: The process '/opt/hostedtoolcache/Python/3.8.6/x64/bin/fontbakery' failed with exit code 1`

## Expected behaviour

(Describe here what you expected to happen, or what you wish was different)

## Resources and exact process needed to replicate

Not sure on that. Here is a link https://github.com/JetBrains/JetBrainsMono/actions

**Comment by felipesanches** [comment, marked as answer]
When your font project has some problem detected by FontBakery, the tool returns a non-zero error code which is often used by other tools such as the GitHub Actions infrastructure to detect that something went wrong. That's why you get a "broken build" in this case. This is the expected behaviour, it is not a bug.

You should look at the GitHub Actions build log to get the full details on what are the problems that FontBakery detected in the latest commit of your font project repo. You will see some **FAIL** messages (highlighted in red), and maybe also some **WARN** messages (in yellow). You must address the FAIL ones and should at least consider what the WARN ones tell you.

Here's one of those **FAIL**s:
https://github.com/JetBrains/JetBrainsMono/runs/1589880289?check_suite_focus=true#step:7:902

![Screenshot from 2020-12-27 10-48-42](https://user-images.githubusercontent.com/213676/103172279-25d14080-4831-11eb-8310-e4edf7895c04.png)

As FontBakery development continues, we sometimes add new checks. So the fact that you did not get these messages before can have two different explanations: **(1)** you may have made changes to your font files that introduced a problem or **(2)** you may have had these problems already but was unaware of it and now FontBakery introduced a check that detects it.

**Reply by philippnurullin** [reply]
Thanks for detailed reply.

**Reply by philippnurullin** [reply]
Understood why i didn't look at the error messages. 
When opening the report in Actions it auto scrolls to the big red line with error text about some problems with in Font Bakery.

**Reply by philippnurullin** [reply]
I have a question on one particular Error:
 > Check:   com.google.fonts/check/family/win_ascent_and_descent
    Desc:    Checking OS/2 usWinAscent & usWinDescent.
    Files:   JetBrainsMono-Light.ttf
    * FAIL: OS/2.usWinAscent value should be equal or greater than 1120, but got 1020 instead [code: ascent]
    * FAIL: OS/2.usWinDescent value should be equal or greater than 401, but got 300 instead. [code: descent]
    Result: FAIL

Looks like it won't be possible to fix this. The height on BoxDrawing element must be height that the WinAsc/Desc. Otherwise in some programming apps there will be gaps (when drawing the Commander interface from type elements) which is unpleasant to work with. What can be done to clear this test and leave the BoxDrawing designed as now?

**Reply by chrissimpkins** [reply]
> I have a question on one particular Error:
> 
> > Check:   com.google.fonts/check/family/win_ascent_and_descent
> > Desc:    Checking OS/2 usWinAscent & usWinDescent.
> > Files:   JetBrainsMono-Light.ttf
> > * FAIL: OS/2.usWinAscent value should be equal or greater than 1120, but got 1020 instead [code: ascent]
> > * FAIL: OS/2.usWinDescent value should be equal or greater than 401, but got 300 instead. [code: descent]
> > Result: FAIL
> 
> Looks like it won't be possible to fix this. The height on BoxDrawing element must be height that the WinAsc/Desc. Otherwise in some programming apps there will be gaps (when drawing the Commander interface from type elements) which is unpleasant to work with. What can be done to clear this test and leave the BoxDrawing designed as now?

You can use the `-x [EXCLUDE_CHECKID]` option one or more times on the command line to exclude individual validations.

**Reply by philippnurullin** [reply]
Thanks for the reply. 
Can you give a hint on where to read about using command line in GitHub actions?
I'm not sure where to begin.

**Comment by chrissimpkins** [comment]
You should be able to update these lines of your GitHub Actions workflows to include the exclusions:

- https://github.com/JetBrains/JetBrainsMono/blob/0cef618578c1e24957620d255285273883015974/.github/workflows/fontbakery-ci-statics.yml#L43
- https://github.com/JetBrains/JetBrainsMono/blob/0cef618578c1e24957620d255285273883015974/.github/workflows/Fontbakery-ci-variable-fonts.yml#L47

The Font Bakery GitHub Action repo + docs are here: https://github.com/f-actions/font-bakery

And the GitHub Action documentation is available here: https://docs.github.com/en/free-pro-team@latest/actions

I made the first draft of the GH Action workflows on your repository and maintain the fontbakery GH Action.  Let me know if you need a hand.
